app/scripts/controllers/detect-tokens.js

```diff
@@ -84,14 +101,23 @@ export default class DetectTokensController {
     if (!this.isActive) {
       return;
     }
-
+    if (
+      process.env.TOKEN_DETECTION_V2 &&
+      (!this.useTokenDetection ||
+        !isTokenDetectionEnabledForNetwork(
+          this._network.store.getState().provider.chainId,
+        ))
+    ) {
+      return;
+    }
     const { tokenList } = this._tokenList.state;
     // since the token detection is currently enabled only on Mainnet
     // we can use the chainId check to ensure token detection is not triggered for any other network
     // but once the balance check contract for other networks are deploayed and ready to use, we need to update this check.
     if (
-      this._network.store.getState().provider.chainId !== MAINNET_CHAIN_ID ||
-      Object.keys(tokenList).length === 0
+      !process.env.TOKEN_DETECTION_V2 &&
```

Comment:
What about the case of `process.env.TOKEN_DETECTION_V2 && Object.keys(tokenList).length === 0` do we need to be returning in that case?

Reply:
This was the existing condition for the previous version, token detection was activated for Mainnet only but the tokenList will be available for other networks as well. This will be removed during flag clean-up.

Reply:
Is it possible for `Object.keys(tokenList).length === 0)` to be true if `process.env.TOKEN_DETECTION_V2` is true?